rts/at/tx/src/main/java/org/jboss/jbossts/star/resource/Transaction.java

```diff
@@ -323,7 +323,9 @@ protected boolean afterCompletion(int arjunaStatus, boolean report_heuristics) {
             }
         }
 
-        coordinator.removeTxState(arjunaStatus, this, enlistmentIds);
+        if (failedList == null || failedList.size() == 0) {
```

Comment:
I would say no because such participants are ignored by the recovery system. 

However, the participant is not allowed to forget the heuristic choice it made until it is told to by the system. This should happen by the standard Narayana recovery mechanisms (ie the REST-AT coordinator does not do it explicitly). Maybe we should consider having a test for that (ie it would verify that the participant is eventually told to forget its decision).